src/rpc/blockchain.cpp

```diff
@@ -1583,13 +1583,17 @@ UniValue savemempool(const JSONRPCRequest& request)
     if (request.fHelp || request.params.size() != 0) {
         throw std::runtime_error(
             "savemempool\n"
-            "\nDumps the mempool to disk.\n"
+            "\nDumps the mempool to disk. It will fail until the previous dump is fully loaded.\n"
             "\nExamples:\n"
             + HelpExampleCli("savemempool", "")
             + HelpExampleRpc("savemempool", "")
         );
     }
 
+    if (!g_is_mempool_loaded && gArgs.GetArg("-persistmempool", DEFAULT_PERSIST_MEMPOOL)) {
```

Comment:
Why is there still an argument check here?  The setting of g_is_mempool_loaded appears to me to be unconditional now.

Reply:
Well, it is now not needed since the g_is_mempool_loaded is set out of the ```if  gArgs.GetArg("-persistmempool"...```. I was thinking that perhaps we should allow to dump even before g_is_mempool_loaded is set if -persistmempool=0 (since in that case it's not loading it anyway).
But perhaps it's better that it waits for that line to be run when -persistmempool=0 too.
 Or perhaps the we can set g_is_mempool_loaded only conditionally like fDumpMempoolLater was.

No strong opinion, I'm happy with any of the possible variations on this. But I don't think it's a matter of documentation. It's just choosing what we want here.

Reply:
Why would there be anything in mempool, when -persistmempool=0 and you are not yet connected to any peers? So there wouldn't be any need to allow `savemempool`.

We are mostly nitpicking here, and I don't care too much what variation you pick, but the current code seems confusion/hard to read, otherwise we woulnd't have this conversation.

Reply:
Alright, I'll just remove the gArgs.GetArg("-persistmempool", DEFAULT_PERSIST_MEMPOOL) condition from here and leave it as it was right before.

Reply:
Fixed.